### PR TITLE
feat: bddtest setup

### PR DESCRIPTION
**Pull request by talwinder50**
closes: #47
Signed-off-by: talwinder.kaur <talwinder.kaur@securekey.com>

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -54,7 +70,30 @@ generate-test-keys: clean
...
+generate-test-keys: clean
+	@mkdir -p -p test/bddtests/fixtures/keys/tls
+	@docker run -i --rm \
+		-v $(abspath .):/opt/go/src/github.com/trustbloc/hub-store \
+		--entrypoint "/opt/go/src/github.com/trustbloc/hub-store/scripts/generate_test_keys.sh" \
+		frapsoft/openssl
+clean:
+	rm -Rf ./.build
+	rm -Rf ./test/bddtests/docker-compose.log
+	rm -Rf ./test/bddtests/fixtures/keys/tls
```
Comment (emirsh) [inline Makefile]: should delete all keys, not just the tls ones

File: scripts/generate_test_keys.sh [hunk 2]
```diff
@@ -9,37 +9,37 @@ set -e
 
 echo "Generating hub-store Test PKI"
 cd /opt/go/src/github.com/trustbloc/hub-store
-mkdir -p test/fixtures/keys/tls
-mkdir -p test/fixtures/keys/server
-mkdir -p test/fixtures/keys/client
+mkdir -p test/bddtests/fixtures/keys/tls
```
Comment (emirsh) [inline scripts/generate_test_keys.sh]: Not sure why this was changed to bddtests... these keys are also used in unit tests

File: test/bddtests/fixtures/docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+#
+# Copyright SecureKey Technologies Inc. All Rights Reserved.
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+version: '2'
+
+services:
+
+  sidetree:
```
Comment (emirsh) [inline test/bddtests/fixtures/docker-compose.yml]: sidetree??

File: scripts/generate_test_keys.sh (hunk 2 above)
Comment (talwinder50) [reply scripts/generate_test_keys.sh]: yea u r right i will put back in fixtures

File: Makefile [hunk 4]
```diff
@@ -20,6 +20,22 @@
 GO_CMD ?= go
 export GO111MODULE=on
 
+# This Makefile assumes a working Golang and Docker setup
+export ARCH=$(shell go env GOARCH)
+DOCKER_CMD         ?= docker
+CONTAINER_IDS = $(shell docker ps -a -q)
```
Comment (llorllale) [inline Makefile]: @talwinder50 this is not required

File: Makefile [hunk 5]
```diff
@@ -20,6 +20,22 @@
...
+DEV_IMAGES = $(shell docker images dev-* -q)
+
+GO_VER             = $(shell grep "GO_VER" .ci-properties |cut -d'=' -f2-)
+
+# Tool commands (overridable)
+DOCKER_CMD ?= docker
```
Comment (llorllale) [inline Makefile]: @talwinder50 this is a duplicate of line 25

File: Makefile [hunk 6]
```diff
@@ -20,6 +20,22 @@
 GO_CMD ?= go
 export GO111MODULE=on
 
+# This Makefile assumes a working Golang and Docker setup
+export ARCH=$(shell go env GOARCH)
+DOCKER_CMD         ?= docker
+CONTAINER_IDS = $(shell docker ps -a -q)
+DEV_IMAGES = $(shell docker images dev-* -q)
```
Comment (llorllale) [inline Makefile]: @talwinder50 not required

File: Makefile [hunk 7]
```diff
@@ -54,7 +70,30 @@ generate-test-keys: clean
...
+	--build-arg ALPINE_VER=$(ALPINE_VER) \
+	--build-arg GO_TAGS=$(GO_TAGS) \
+	--build-arg GOPROXY=$(GOPROXY) .
+
+bddtests: clean checks generate-test-keys hubstore-docker
+	@scripts/integration.sh
 
-all: checks unit-test
+generate-test-keys: clean
+	@mkdir -p -p test/bddtests/fixtures/keys/tls
```
Comment (llorllale) [inline Makefile]: @talwinder50 nitpick: `-p` twice

File: .ci-properties [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+
+if [ $IS_RELEASE == false ]
+then
+  EXTRA_VERSION=snapshot-$(git rev-parse --short=7 HEAD)
+  PROJECT_VERSION=$BASE_VERSION-$EXTRA_VERSION
+else
+  PROJECT_VERSION=$BASE_VERSION
+fi
+
+export HUB_STORE_TAG=$ARCH-$PROJECT_VERSION
```
Comment (llorllale) [inline .ci-properties]: @talwinder50 `HUB_STORE_TAG` is not used elsewhere, which means everything before it is not needed

Comment (talwinder50) [reply .ci-properties]: removed

**Pull request merged**